src/app/features/course-info/course-info.component.ts

```diff
@@ -1,13 +1,34 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnInit } from '@angular/core';
 import { Course } from '@shared/models/course.model';
 import { ButtonText } from '@shared/components';
+import { ActivatedRoute, Router } from '@angular/router';
+import { CoursesStoreService } from '@app/services/courses-store.service';
 
 @Component({
   selector: 'app-course-info',
   templateUrl: './course-info.component.html',
   styleUrls: ['./course-info.component.scss'],
 })
-export class CourseInfoComponent {
-  @Input() course!: Course;
+export class CourseInfoComponent implements OnInit {
+  @Input() course!: Course | null;
   protected readonly ButtonText = ButtonText;
+
+  constructor(
+    private route: ActivatedRoute,
+    private routes: Router,
```

Comment:
I would name it router instead of routes.

Reply:
Yeah, probably misspelled it